### DISCUSSION TITLE
Extism & WebAssembly Component Model

### DISCUSSION THREAD

**Discussion by christophwille** [extism/extism#334] [General]
Is there a plan how Extism will change/adapt/make use of once the Component model is available?

**Comment by esoterra** [comment]
Wasm Components should be available at a "preview 2" level later this year. Since Components offer a standards-path answer to creating high-level interfaces between Wasm code and can start slowing down/reversing fragmentation in the existing solutions, it'd be great to see Extism join that effort and start supporting Components later this year.

**Comment by nilslice** [comment]
We currently don't have concrete plans to make use of the Component Model yet. Our ABI is designed to be as generic as possible to:
1). support all the language environments we work with today, and make it simple to add support for more
2). have a low-level "bring your own serialization" bytes-in/bytes-out model. 

The Component Model could certainly be useful, but until it catches up a bit in terms of language support, we are going to continue on our path. 

We will have a version 1.0 roadmap available soon, and will share with our community on GitHub and Discord. Please join us if you're not already there: https://discord.gg/cx3usBCWnc
